### PR TITLE
Adjust workaround for Windows precompilation on Julia 1.8.2+

### PR DESCRIPTION
**Pull request by BioTurboNick**
My goal here is to still precompile on normal computers, but only skip it if there's a problem in CI

Providing a warning will also allow the issue to be monitored, and if someone ever encounters it outside of the CI system, maybe we can learn more about the issue.

Adjustment to #4406, based on work in #4445

### CODE REVIEW

**Comment by BioTurboNick** [comment]
This works to avoid the CI error but the warning doesn't show up...

**Comment by t-bltg** [comment]
I'm a bit reluctant to use `try catch`, but if there is not other solution :shrug: 

Can you check that this has no impact on `TTFP` by running [this snippet](https://github.com/JuliaPlots/Plots.jl/pull/4334#issuecomment-1250745773) against this PR and master ?

For this `PR` to go in, it's mandatory that the warning shows up somehow ...

**Comment by BioTurboNick** [comment]
Will do, though my expectation is that try/catch has 0 cost *unless* an exception occurs.

Also will see if there's a way to get the warning to show up...

**Comment by BioTurboNick** [comment]
Hmm. Tried a few things and no way so far to get precompilation to print anything unless there's an error thrown?

Inquiring here: https://github.com/JuliaLang/Pkg.jl/issues/3224

File: src/precompilation.jl [hunk 1]
```diff
@@ -5,25 +5,35 @@ if get(ENV, "PLOTS_PRECOMPILE", "true") == "true"
...
-                $func()
-            end)
+                    try
+                        # During precompilation on the Windows GitHub CI in Julia 1.8.2+,
+                        # can fail here due to GR potentially failing to write out the
+                        # temporary file. Pending a fix, swallow the error and continue.
+                        $func()
+                    catch
+                        @debug "Plots: Failed a trial save during precompilation"
+                    end
```
Comment (t-bltg) [inline src/precompilation.jl:34]:
```suggestion
                        $func()
                    catch e
                        if e isa SystemError && Sys.iswindows() && get(ENV, "CI", "false") == "true"
                            # During precompilation on the Windows GitHub CI in Julia 1.8.2+,
                            # can fail here due to GR potentially failing to write out the
                            # temporary file. Pending a fix, swallow the error and continue.
                        else
                            throw(e)
                        end
                        @debug "Plots: Failed a trial save during precompilation"
                    end
```

If the warning doesn't work as expected, then maybe silently ignore precompilation `SystemEror` on `windows` `ci`, else hard error.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,7 +47,19 @@ jobs:
...
+          JULIA_PKG_PRECOMPILE_AUTO: 0
+      
+      # surface errors during precompilation
+      # auto-precompile in build step suppresses errors currently.
+      - name: Precompile
+        env:
+          JULIA_DEBUG: Plots
+        shell: julia --project=@. --color=yes {0}
+        run: |
+          using Pkg; Pkg.precompile()
```
Comment (t-bltg) [inline .github/workflows/ci.yml:62]:
```suggestion
```

That does not seem to work as intended right ? No warning appear on `windows 1.8.2`.

Comment (BioTurboNick) [reply .github/workflows/ci.yml:62]: Yes, at the moment it does not. Was hoping for another workaround from the Pkg issue.

File: src/precompilation.jl (hunk 1 above)
Comment (BioTurboNick) [reply src/precompilation.jl:34]: That's fair. We probably want `rethrow()` though?

Comment (t-bltg) [reply src/precompilation.jl:34]: Ha, yes I took `rethrow(e)` note in https://docs.julialang.org/en/v1/base/base/#Base.rethrow too seriously.

**Comment by t-bltg** [comment]
Superseded by https://github.com/JuliaPlots/Plots.jl/pull/4617.

**Pull request closed without merging**